yann/utils/image.py

```diff
@@ -45,7 +51,7 @@ def gray2rgb(r, g, b, channels_dim = 3):
     if channels_dim == 3:
         return out
     elif channels_dim == 1:
-        return out.transpose((0,3,1,2))
+        return out.transpose((2,0,1))
```

Comment:
The code couple of lines before this asserts if the number of dimensions in r,g,b are 2. We are adding another dimension to it by concatenating them. In total we have 3 dimension with channels as the third one with index 2 ie., B01C format. So to get the channel at 1 i.e., convert it to BC01 format. we need to do give 2,0,1 to transpose such that the first dimension is channels.